### PR TITLE
lib/vfscore: Add additional flags for compatibility

### PR DESCRIPTION
**Pull request by jschlumpp**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): N/A

### Additional configuration

 - `CONFIG_LIBVFSCORE=y`

### Description of changes

This adds some additional flags which can be used for filesystem implementations.

### CODE REVIEW

**Review by razvand — COMMENTED** [review]
Hi, @mschlumpp . What was the use case for these flags? I didn't find `IO_ASYNC` nor `IO_DIRECT` in Linux. Where did you adapt the values for them?

**Comment by jschlumpp** [comment]
They are from FreeBSD and are necessary for the ext filesystem driver. The values don’t matter as much because only the symbols are used.

File: lib/vfscore/include/vfscore/vnode.h [hunk 1]
```diff
@@ -144,6 +144,8 @@ struct vattr {
 
 #define IO_APPEND	0x0001
 #define IO_SYNC		0x0002
+#define IO_ASYNC	0x0004
+#define IO_DIRECT	0x0008
```
Comment (StefanJum) [inline lib/vfscore/include/vfscore/vnode.h]: Please add some short comments on these macros, like the one on the `MNT_NOATIME`.

**Review by StefanJum — CHANGES_REQUESTED** [review]
Only one comment.

Other than that, does it make sense to add the flags if we are not using them at the moment, or should we add them when they will be used.

**Comment by razvand** [comment]
@mschlumpp, the reason I'm concerned about the actual values is in case of a binary compatibility issue. If the symbols / values are only going to be used internally in Unikraft, then it's good.

I understand @StefanJum's comment on keeping this PR open until we have an actual use (i.e. a PR that makes use of these macros). I think this is something that makes sense long term. The addition of a feature, function, macro, should be supported by a use case.

**Comment by jschlumpp** [comment]
The `IO_` constants are only relevant for "kernel"-level interactions. Therefore, ABI compatiblity for userspace applications is not a concern.

**Comment by razvand** [comment]
@StefanJum, we can approve this PR. Please review / approve it, and then I'll do it.

**Review by StefanJum — APPROVED** [review]
All good, thanks.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**